strawberryfields/apps/similarity.py

```diff
@@ -510,21 +652,98 @@ def feature_vector_mc(
         loss (float): fraction of photons lost in GBS
 
     Returns:
-        list[float]: a feature vector of event probabilities in the same order as
-        ``event_photon_numbers``
+        list[float]: a feature vector of orbit probabilities
     """
-    if samples < 1:
-        raise ValueError("Number of samples must be at least one")
+    if len(event_photon_numbers) <= 0:
+        raise ValueError("List of photon numbers must have at least one element")
+    if min(event_photon_numbers) < 0:
+        raise ValueError("Cannot request events with photon number below zero")
+    if max_count_per_mode < 0:
+        raise ValueError("Maximum number of photons per mode must be non-negative")
     if n_mean < 0:
         raise ValueError("Mean photon number must be non-negative")
     if not 0 <= loss <= 1:
         raise ValueError("Loss parameter must take a value between zero and one")
+
+    if samples:
+        return [
+            prob_event_mc(graph, photon_number, max_count_per_mode, n_mean, samples, loss)
+            for photon_number in event_photon_numbers
+        ]
+
+    return [
+        prob_event_exact(graph, photon_number, max_count_per_mode, n_mean, loss)
+        for photon_number in event_photon_numbers
+    ]
+
+
+def feature_vector_orbits_sampling(samples: list, list_of_orbits: list) -> list:
+    r"""Calculates feature vector of given orbits with respect to input samples.
+
+    The feature vector is composed of orbit probabilities reconstructed by measuring the
+    occurrence of given orbits in the input ``samples``.
+
+    **Example usage:**
+
+    >>> from strawberryfields.apps import data
+    >>> samples = data.Mutag0()
+    >>> feature_vector_orbits_sampling(samples, [[1,1], [2], [2,1,1], [1,1,1,1]])
+    [0.19035, 0.0, 0.05175, 0.1352]
+
+    Args:
+        samples (list[list[int]]): a list of samples
+        list_of_orbits (list[list[int]]): a list of orbits
+
+    Returns:
+        list[float]: a feature vector made up of estimated orbit probabilities

```

Comment:
In the return part of `feature_vector_events_sampling`, we mention the order - would be good to also do it here.